rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py

```diff
@@ -135,8 +136,9 @@ def defragment(self, message):
             ):
             self.received_fragments[msg_id]["fragment_list"][msg_num] = msg_data
             self.received_fragments[msg_id]["timestamp_last_append"] = now
-            log_msg = "appended fragment #" + str(msg_num)
-            log_msg += " (total: " + str(msg_total)+ ") to fragment list for messageID " + str(msg_id)
+            log_msg = ["appended fragment #" + str(msg_num)]
+            log_msg.append(" (total: " + str(msg_total)+ ") to fragment list for messageID " + str(msg_id))
+            log_msg = ''.join(log_msg)
```

Comment:
For cases like this one, where we're not appending anything conditionally or in a loop, Python string formatting is equally as fast and (in my opinion) a bit easier to read:
```
self.protocol.log(
    'debug',
    'appended fragment #{:d} (total: {:d}) to '
    'fragment list for messageID {:s}'.format(
        msg_num,
        msg_total,
        msg_id)
```